FindBounce/Kernel/FindBounce.m

```diff
@@ -1696,60 +1725,54 @@
 	If[noFields==1,maxItePath = 0];
 	(*Bounce and path deformation.*)
 	While[iter <= maxItePath||switchPath,
-		
-		(*Rule.*)
-		rule = Table[fields[[i]]->fieldPoints[[s,i]],{s,Ns+1},{i,noFields}];
-		(* TODO: Eventually move this intermediate calculation inside each major function below. *)
-		\[Phi]L=longitudinalProjection[fieldPoints];
-		eL=unitVectors[fieldPoints];
-		l=segmentsLength[fieldPoints];
-		
+	
 		(*Single Field Bounce.*)
-		{actionP,VL,v,a,b,pos,R,initialR} = If[
+		{actionP,potentialValues,bounce,initialR} = If[
 			Not[bottomless],
-			SingleFieldBounce[V,potentialPoints,Ns,noFields,\[Phi]L,dim,maxIteR,
-				actionTolerance,ansatzInitialR,initialR,rule,iter,
-				switchPath||iter==maxItePath
+			SingleFieldBounce[potentialValues,fieldPoints,
```

Comment:
Nice! This looks clearer and simpler to me.

Reply:
I liked it as well. You had a good point. From now on I will try to have only functions with few arguments.